### ISSUE TITLE
fatal: Invalid revision range

### ISSUE THREAD

**Issue by jasonkarns** [redhat-plumbers-in-action/differential-shellcheck#586]
### Type of issue

Bug Report

### Description

Action is reporting:

```
fatal: Invalid revision range e7d05fd3533403e8aa79d4c22b31d106ec3877ac..cdb538f5c530addbc82c353a94e5d29844d043d4
Warning:  Please check if the repository was cloned with `fetch-depth: 0`. Differential ShellCheck needs the entire history to work correctly. 
```

But the action is indeed configured with full history clone:

```
  shellcheck:
    permissions: { contents: read, security-events: write }
    runs-on: ubuntu-latest
    steps:
      - uses: step-security/harden-runner@9af89fc71515a100421586dfdb3dc9c984fbf411 # v2.19.4
        with: { egress-policy: audit }
      - uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
        with: { fetch-depth: 0 }
      - uses: redhat-plumbers-in-action/differential-shellcheck@d965e66ec0b3b2f821f75c8eff9b12442d9a7d1e # v5.5.6
        with:
          severity: error # TODO strengthen
          token: ${{ secrets.GITHUB_TOKEN }}
```

Error is in public repo so the failed check is public: https://github.com/nodenv/node-build/actions/runs/27300844352/job/80645929835?pr=1105

**Comment by jasonkarns** [comment]
This is still occurring regularly.

**Comment by jamacku** [comment]
@jasonkarns I have prepared PR that should address this issue. https://github.com/nodenv/node-build/pull/1121

**Comment by jasonkarns** [comment]
This doesn't seem to address the bug, right? I wouldn't expect the action to be limited to only specific branches. The expected configuration would allow the check to run on _any_ pushed branch (which is why the initial configuration omitted a branch restriction).

Similarly, users should be able to open PRs with _any_ targeted base branch. And the shellcheck should still be running on any PR, not just a limited set.
